### DISCUSSION TITLE
请问下路由规则接口里  uri/uris，host/hosts对路由匹配速度会有影响吗？

### DISCUSSION THREAD

**Discussion by liuguangyu133** [apache/apisix#7878] [Q&A]
在给apisix做接口封装的时候发现路由规则当中如果是单个路径使用的是 uri(string)，多个路径的时候是uris(list)；host字段也是同理。
所以我们封装的时候想不做判断，即使只有一条uri也放到uris里，例如  uris: ['/api'] 
因此想了解下 uri/uris 和 host/hosts这样设计有没有其他什么原因，比如会对路由匹配的速度造成影响之类？

**Comment by tzssangglass** [comment]
No impact, so designed for ease of use. `uris` will be traversed as a single `uri` to create a radixtree. ref: https://github.com/api7/lua-resty-radixtree/blob/master/lib/resty/radixtree.lua#L409-L416
